### PR TITLE
Make lookahead rely on native pytorch Conv1d

### LINKED ISSUE #274: LookAhead layer vs Bi-rnn performance comparision
Hi,
these are the two models i am training:
1. 3 X Bi-rnn , 2560 nodes  ==> Number of parameters: 66040608
2. 3 X rnn , 2560 nodes , lookahead layer with context 20. ==> Number of parameters: 33244448

machine config:
12 vCPUs, 64 GB memory
4 x NVIDIA Tesla P100

training on a distributed data parallel set up:
in scenario 1, though the number of parameters is more i am able to use a **batch size of 45** but in model 2 i am able to use only a **batch size of 10**.
My question is - 
Why is the use of LookAhead layer causing a drastic reduction in batch size?
(My understanding is the lookAhead layer is introduced to increase the training speed but i see a contradiction here)

### PR DESCRIPTION
**Pull request by Rasmusafj**
Currently, the lookahead implementation increases the memory required to train models. In my case, I had to use a batch size of 16 instead of 32 to avoid memory issues. 

This PR fixes the memory issues by performing the row convolutional operation (lookahead) with pytorch's Conv1D layer. 

I'm currently running tests to see whether it is beneficial to also include batch normalization and/or bias to the lookahead layer with a private dataset. Will update this PR if that is the case. 

Please tell me if you spot something being off with the improvement.

Fixes #274

### CODE REVIEW

**Comment by SeanNaren** [comment]
This is awesome! Thank you!
